vaadin-rich-text-editor-flow-parent/vaadin-rich-text-editor-flow/src/main/java/com/vaadin/flow/component/richtexteditor/RichTextEditor.java

```diff
@@ -169,6 +169,18 @@ public void setValueChangeMode(ValueChangeMode valueChangeMode) {
      * <p>
      * Note: {@link Binder} will take care of the {@code null} conversion when
      * integrates with the editor, as long as no new converter is defined.
+     * <p>
+     * Since v24, this method only accepts values in the HTML format, whereas in
+     * v23 and earlier this method would accept values in the Delta format. In
+     * order to prevent data corruption, passing a value that starts with either
+     * <code>[</code> or <code>{</code> will now throw an
+     * {@link IllegalArgumentException}, as it might indicate that the value is
```

Comment:
I wonder if this is too sneaky. If someone doesn't notice this comment, the exception might occur only until in production.

Reply:
There was a long discussion about this solution in this [internal Slack thread](https://vaadin.slack.com/archives/C3TGRP4HY/p1674637523802949). Some alternatives, like changing the API so that it wouldn't compile anymore, were considered, but we ultimately decided to go with the exception.